include/net/nrf_cloud_pgps.h

```diff
@@ -239,6 +239,34 @@ int nrf_cloud_pgps_request_all(void);
 #endif /* CONFIG_NRF_CLOUD_MQTT */
 
 #if defined(CONFIG_NRF_CLOUD_PGPS_TRANSPORT_NONE)
```

Comment:
The way we are using `CONFIG_NRF_CLOUD_PGPS_TRANSPORT_NONE` it almost seems like it would be better to define a mutually exclusive set of kconfig values:

* `CONFIG_NRF_CLOUD_PGPS_TRANSPORT_NONE`
* `CONFIG_NRF_CLOUD_PGPS_TRANSPORT_CUSTOM`
* `CONFIG_NRF_CLOUD_PGPS_TRANSPORT_WHATEVER_ELSE`

and use `CONFIG_NRF_CLOUD_PGPS_TRANSPORT_CUSTOM` everywhere you are using `NONE` in this PR

I'm not sure what `NONE` is being used for outside this PR, maybe it should just be renamed to `CUSTOM`

Reply:
I only implemented the MQTT transport.  Someone else added NONE to support REST outside of the library.  Not sure of the reasoning.  I added this new stuff to enable the LWM2M team to make progress.

I'll think about your ideas when I get back to this PR.

Reply:
In case it is helpful for when you return to this PR:

It looks like NONE was added here:
https://github.com/nrfconnect/sdk-nrf/pull/5111

Brilliantly, the commit logs explain the reasoning:

> Currently, the only supported transport
is MQTT. The alternative is to select "none" as transport, and
instead receive PGPS_EVT_REQUEST events when a request would
otherwise have been sent if MQTT was enabled.